lib/data_fifo/data_fifo.c

```diff
@@ -167,6 +167,22 @@ int data_fifo_empty(struct data_fifo *data_fifo)
 	return 0;
 }
 
+int data_fifo_deinit(struct data_fifo *data_fifo)
+{
+	__ASSERT_NO_MSG(data_fifo != NULL);
```

Comment:
Recommend using ASSERT with message.

Reply:
On this, It is good to test like this, but if there is an issue in the release that would have exercised this assert but it has been optimised out, then the codes action would be undetermined and potentially difficult to debug. Would be more code but better in my option to explicitly test for NULL. Actually it shouldn't be more than a test zero and branch (with branch prediction this will be even better). An error is already returned and tested for, so the code could gracefully fail, rather than go off somewhere into memory!